### PR TITLE
Kconfig: Expose pkg/wakaama configurations

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This PR moves the LwM2M Wakaama client configuration macros to the `CONFIG_` namespace and exposes them to Kconfig.

### Testing procedure
- The Wakaama example should work as usual
- You should be able to configure the client by running `make menuconfig`. Test e.g. changing the server URI and device name, enabling the package logs, and enabling the usage of a bootstrap server.

### Issues/PRs references
Part of #12888
~~Depends on #12913~~

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
Rebased to master, this has no dependencies now.

File: pkg/wakaama/include/lwm2m_client_config.h [hunk 1]
```diff
@@ -79,110 +79,161 @@ extern "C" {
...
+ *
+ * This options are meant to be set either via Kconfig or CFLAGS:
  *
- * Valid values are:
- * - U: UDP
- * - UQ: UDP with Queue mode
- * - S: SMS
- * - SQ: SMS with Queue mode
- * - US: UDP and SMS
- * - UQS: UDP with Queue mode and SMS
```
Comment (fjmolinas) [inline pkg/wakaama/include/lwm2m_client_config.h:154]: Did you remove these because now they are defined in kconfig?

Comment (fjmolinas) [reply pkg/wakaama/include/lwm2m_client_config.h:154]: Oh got it, sorry for the noise.

File: examples/wakaama/Makefile [hunk 2]
```diff
@@ -31,19 +31,17 @@ DEVELHELP ?= 1
...
 # Uncomment to indicate that the server is a LwM2M bootstrap server
-# CFLAGS += -DLWM2M_BOOTSTRAP=1
-
-# NOTE: Use wakaama in client mode
-CFLAGS += -DLWM2M_CLIENT_MODE
+# CFLAGS += -DCONFIG_LWM2M_BOOTSTRAP=1
 
 include $(RIOTBASE)/Makefile.include
+
+# Configure server via CFLAGS only is not done via Kconfig
```
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply examples/wakaama/Makefile]: Changed

**Comment by fjmolinas** [comment]
Symbol changes are good and tested configuration changes take effect:

image

**Comment by leandrolanzieri** [comment]
> I think documenation wise it would be good to add an a mention to the fact that this can all be set through `kconfig`/ `make menuconfig`

I added some improvements to the `README.md`, also indicating how to bind the host on Leshan

**Comment by fjmolinas** [comment]
@leandrolanzieri looks good now, please squash!

**Comment by fjmolinas** [comment]
GO!

**Pull request merged**